### PR TITLE
Use purescript-github-actions-toolkit library

### LINKED ISSUE #4: Split out bindings to GitHub Actions toolkit as a new library
The core of this library is built on top of the [GitHub Actions toolkit](https://github.com/actions/toolkit), a set of libraries for executing commands within GitHub Actions. It is likely that others will want to write GitHub Actions in PureScript in the future, and these bindings are broadly useful outside this library.

Some possible actions that come to mind which could use this shared base:

1. Automatically format code using `purty` 
2. Automated attempts to upgrade package sets / dependencies in PureScript repositories
3. Automatically upload Pursuit documentation for releases which haven't been uploaded yet

All of the `Actions*` modules that exist in this library can be moved directly out to another package that exposes something like a set of modules named:

```
GitHub.Actions.ToolCache
GitHub.Actions.Cache
GitHub.Actions.GitHub
GitHub.Actions.IO
GitHub.Actions.Exec
...
```

which mirror the packages available in the official toolkit. The bindings weren't difficult to put together in the first place, so this shouldn't be more than a few hours' effort to set up.

### PR DESCRIPTION
**Pull request by colinwahl**
* Removes the `src/Actions` tree
* Adds a the `github-actions-toolkit` dependency
* Implements `setup-purescript` using the external bindings

This intends to close #4

### CODE REVIEW

**Comment by colinwahl** [comment]
I'm not sure how to get CI to run on this branch - I have run it on a private copy I have of this repo, but ideally I'd be able to run it here as well.

**Comment by colinwahl** [comment]
I created https://github.com/purescript-contrib/setup-purescript/pull/9 which I believe will make CI run on the commits to future PRs

**Comment by thomashoneyman** [comment]
Closes #4 also.

File: packages.dhall [hunk 1]
```diff
@@ -23,3 +23,21 @@ in  upstream
     , version =
         "v5.0.1"
     }
+
+  with github-actions-toolkit =
```
Comment (thomashoneyman) [inline packages.dhall]: Since this is the latest package set, mind running spago upgrade-set?

File: src/Main.purs [hunk 2]
```diff
@@ -2,18 +2,27 @@ module Main where
...
 import Setup.BuildPlan (constructBuildPlan)
 import Setup.GetTool (getTool)
 import Setup.UpdateVersions (updateVersions)
 
 main :: Json -> Effect Unit
-main json = do
-  tools <- constructBuildPlan json
-  launchAff_ $ traverse_ getTool tools
+main json = runAff_ go $ runExceptT do
+  tools <- ExceptT (liftEffect (runExceptT (constructBuildPlan json)))
```
Comment (thomashoneyman) [inline src/Main.purs]: Is this mapExceptT liftEffect?

**Review by thomashoneyman — COMMENTED** [review]
Once tests run I’m in! Thanks!

File: packages.dhall (hunk 1 above)
Comment (colinwahl) [reply packages.dhall]: Ah! Didn't know how 😅

File: src/Main.purs (hunk 2 above)
Comment (colinwahl) [reply src/Main.purs]: Oh yeah! Knew I was forgetting something

**Comment by colinwahl** [comment]
Aha! Tests are running :)

**Review by thomashoneyman — APPROVED** [review]
Might be nice to link to these bindings in the README, as they’re worth promoting.

**Pull request merged**